core/src/components/select/select.ionic.scss

```diff
@@ -96,6 +96,10 @@
   width: 100%;
 }
 
+:host(.select-label-placement-stacked) .select-wrapper-inner{
+  flex-grow: 0;
+}
```

Comment:
This approach does not align with the styling structure. `flex-grow` is being applied to all the themes because it exists within the common file. Since `ionic` theme does not require this style, then we need to move the following styles to the `select.native.scss` file: https://github.com/ionic-team/ionic-framework/blob/6e58dbd78321b8013bb637fb423b5fa10b4b3ffd/core/src/components/select/select.common.scss#L224-L233

Once that is moved then the styles that were added to the `ionic` theme can be removed. Plus, I would recommend moving the following styles to `select.native.scss` file as well:
https://github.com/ionic-team/ionic-framework/blob/6e58dbd78321b8013bb637fb423b5fa10b4b3ffd/core/src/components/select/select.common.scss#L60-L68

I would recommend reviewing how the file structure are set for the themes. That way we're not having to overwrite styles when we can move them to the appropriate locations.